### ISSUE TITLE
Provide a full list of resolved collection-level dependencies via API

### ISSUE THREAD

**Issue by shanemcd** [ansible/galaxy_ng#242]
### Use Case

This will be needed by [ansible-builder](https://github.com/ansible/ansible-builder) in order to construct a list of python / system level dependencies. We will need to inspect each collection for the information surfaced by https://github.com/ansible/galaxy_ng/issues/23

### Proposed Solution

I'm unopinionated on the implementation details, but if it helps to know what I'm currently doing... the only way for me to get the data I need is via a private method in the Galaxy CLI:

```python
from ansible import context
from ansible.galaxy import Galaxy
from ansible.galaxy.api import GalaxyAPI
from ansible.galaxy.collection import _build_dependency_map


context.CLIARGS = {'type': 'collection', 'ignore_certs': True}
galaxy = Galaxy()
api = GalaxyAPI(galaxy, 'default', 'https://galaxy.ansible.com')
dependency_map = _build_dependency_map([('alancoding.deps', '*', None)],
                                       [],
                                       '/tmp/test',
                                       [api], False, False, False, False)
print(dependency_map)
```

**Comment by AlanCoding** [comment]
Note: the above script works in Ansible 2.9 but fails in `devel` with

```python
Traceback (most recent call last):
  File "testing/shane.py", line 13, in <module>
    [api], False, False, False, False)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/galaxy/collection.py", line 1131, in _build_dependency_map
    for name, version, source, req_type in collections:
ValueError: not enough values to unpack (expected 4, got 3)
```

**Comment by chouseknecht** [comment]
Subtask or further definition of #23

**Comment by AlanCoding** [comment]
Here's an example of what the current MANIFEST information looks like, by doing `ansible-galaxy collection install community.aws` and then `cat ~/.ansible/collections/ansible_collections/community/aws/MANIFEST.json`

```json
  "dependencies": {
   "amazon.aws": ">=0.1.0"
  },
```

This is for the case that the installed `community.aws` collection is version 1.0.0.

The Galaxy API has models for the collection itself, as well as for collection versions. For the above example:

https://galaxy.ansible.com/api/v2/collections/community/aws/versions/1.0.0/

This reports the same information as in the `MANIFEST.json`, including the version range for `amazon.aws`. Starting an un-versioned collection specified by the user, it's possible to find the most recent version of that collection, and then to find the un-versioned collection-level requirements of that from the Galaxy API (for collections which are in Galaxy to begin with).

Now, assume we have a desired version of a desired collection, 1.0.0 of `community.aws`. The resolved set of dependencies is not static. Tomorrow, `amazon.aws` may release version 0.1.1, which would satisfy its dependencies.

To summarize:

 - the Galaxy API already gives un-resolved collection-level dependencies
 - resolved dependencies for a given collection version is essentially a living document, not static in time

It is possible to use the Galaxy data to provide a resolved set of dependencies (doing dependency resolution logic), either working through the API, or doing the logic in the backend itself.

I want to be upfront about this early, because if, hypothetically, an endpoint `/api/v2/collections/community/aws/versions/1.0.0/resolved_dependencies/` is introduced, this would be a product of server-side logic and subject to change, as new content is introduced.

**Comment by newswangerd** [comment]
Moved to issues.redhat.com/
